### ISSUE TITLE
Factor out EC2 instance state checking logic into its own module

### ISSUE THREAD

**Issue by opadron** [Kitware/gobig#53]
Follow up from @kotfic's [comment](https://github.com/Kitware/gobig/pull/52#issuecomment-191817773) in #52.

We should pull out the code in the [ec2 filter plugin](https://github.com/opadron/gobig/blob/ec2-pod-rework/filter_plugins/ec2.py) that queries AWS for the instance states into its own module.  Such a stand-alone module would be useful for many cases outside of using the `ec2-pod` roles.

**Comment by kotfic** [comment]
@opadron @danlamanna Dan and I talked about this some today actually,  we are thinking we're going to look at this from a dynamic inventory approach rather than a module.  This may still be useful,  but it isn't a high priority for us atm.

**Comment by opadron** [comment]
Yeah, the idea with this issue is basically the logical extension of the idea of managing playbook inventory as part of the play (instead of through a separate inventory file).  I figured such a proposed module would basically work like the `ec2.py` ansible inventory script, except in module form.

I still think this is worth doing, but we can put it in the back burner if it's not a high-priority item.

**Comment by kotfic** [comment]
Yeah,  there is no reason it couldn't act as a module and a dynamic inventory script (seeing as the code would be 90% the same). But as is we just need the dynamic inventory script.  If someone comes back to this later they should touch base with @danlamanna to see what parts of the code he writes for cumulus could be reused to implement this enhancement
